File: 3626-smallest-divisible-digit-product-i/3626-smallest-divisible-digit-product-i.py
class Solution(object):
    def smallestNumber(self, n, t):
        """
        :type n: int
        :type t: int
        :rtype: int
        """
        while True:
            num = n
            pro = 1
            while num:
                pro *= (num % 10)
                num //= 10
            if pro % t == 0:
                return n
            n += 1
        

File: 3626-smallest-divisible-digit-product-i/3626-smallest-divisible-digit-product-i.cs
public class Solution {
    public int SmallestNumber(int n, int t) {
        for (; ; n++) {
            var num = n;
            var pro = 1;
            while (num != 0) {
                pro *= (num % 10);
                num /= 10;
            }
            if (pro % t == 0)
                return n;
        }
        return -1;
    }
}